Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -0,0 +1,877 @@
+                        $nonCompliantMDCTierResourcetype += $resource 
+                    }
+
+                    if ($EnableKeyVault -eq $true -and $_.Name -eq "KeyVaults") {
+                        $nonCompliantMDCTierResourcetype += $resource 
+                    }
+        }
+    }
+   
+    $nonCompliantMDCTypeCount = ($nonCompliantMDCTierResourcetype | Measure-Object).Count
+
+    # If control is already in Passed state (i.e. 'Microsoft.Security' provider is already registered and no non-compliant resource types are found) then no need to execute below steps.
+    if($isProviderRegister -and ($nonCompliantMDCTypeCount -eq 0))
+    {
+        Write-Host "[$($reqProviderName)] provider is already registered and there are no non-compliant resource types. In this case, remediation is not required."  -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        return
+    }
+
+    Write-Host "Found [$($nonCompliantMDCTypeCount)] resource types without [$($reqMDCTier)] pricing tier."
+
+    $colsProperty =  @{Expression = { $_.Name }; Label = "Name"; Width = 40; Alignment = "left" },
+            @{Expression = { $_.PricingTier }; Label = "PricingTier"; Width = 40; Alignment = "left" },
+             @{Expression = { $_.Id }; Label = "Id"; Width = 80; Alignment = "left" }
+       
+       
+
+    $nonCompliantMDCTierResourcetype | Format-Table -Property $colsProperty -Wrap
+    
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up resource type details..."
+    Write-Host $([Constants]::SingleDashLine)
+   
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\AzureDefender"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    # Backing up resource type details.
+    $backupFile = "$($backupFolderPath)\NonCompliantResourceTypes.csv"
+    $nonCompliantMDCTierResourcetype | Export-CSV -Path $backupFile -NoTypeInformation
+
+    Write-Host "Resource type details have been backed up to" -NoNewline
+    Write-Host " [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+
+    # Performing remediation
+    if($nonCompliantMDCTypeCount -gt 0)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4]  Remediating non compliant resource type..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+
+        if (-not $Force)
+        {
+            Write-Host "This step will remediate non compliant resource type for subscription [$($context.Subscription.SubscriptionId)]" -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
make non compliant as non-compliant at every place

Reply:
atlst in messages which is shown to user

Reply:
Done.